server/src/main/java/org/apache/druid/metadata/SegmentsMetadataManager.java

```diff
@@ -144,6 +144,12 @@ Optional<Iterable<DataSegment>> iterateAllUsedNonOvershadowedSegmentsForDatasour
    */
   List<Interval> getUnusedSegmentIntervals(String dataSource, DateTime maxEndTime, int limit);
```

Comment:
Is there a reason not to get rid of this function from the interface? It looks like there are no callers other than the KillUnusedSegmentsDuty

Reply:
done